pandas/tests/frame/test_misc_api.py

```diff
@@ -61,37 +61,32 @@ def test_join_index(self):
         # left / right
 
         f = self.frame.reindex(columns=['A', 'B'])[:10]
-        f2 = self.frame.reindex(columns=['C', 'D'])
+        f2 = self.frame.reindex(columns=['C', 'D'])[5:][::-1]
 
         joined = f.join(f2)
         self.assert_index_equal(f.index, joined.index)
-        self.assertEqual(len(joined.columns), 4)
+        expected_columns = pd.Index(['A', 'B', 'C', 'D'])
+        self.assert_index_equal(joined.columns, expected_columns)
```

Comment:
Sorry, I was talking about the columns, not index. I think the columns of the result are not influenced by inner vs outer join?